spec/controllers/snipets_controller_spec.rb

```diff
@@ -1,174 +1,140 @@
-        put :update, {:id => snipet.to_param, :snipet => valid_attributes}
-        expect(response).to redirect_to(snipet)
+    end
+    context "the snippet is not valid" do
+      before{expect(generic_snippet).to receive(:update).and_return(false)}
+      it "render template edit" do
+        expect(put :update, :id => generic_snippet.id, :snipet => generic_snippet.attributes).to render_template(:edit)
       end
     end
+  end
 
-    context "with invalid params" do
-      it "assigns the snipet as @snipet" do
-        sign_in build(:user)
-        snipet = Snipet.create! valid_attributes
-        put :update, {:id => snipet.to_param, :snipet => invalid_attributes}
-        expect(assigns(:snipet)).to eq(snipet)
+  describe "DELETE #destroy" do
+    context "the snippet exists" do
+      before{expect(generic_snippet).to receive(:destroy).and_return(true)}
+      it "call delete on the snippet and redirect_to all snippets" do
+        expect(delete :destroy, :id => generic_snippet.id).to redirect_to(:action => :index)
       end
-
-      it "re-renders the 'edit' template" do
-        sign_in build(:user)
-        snipet = Snipet.create! valid_attributes
-        put :update, {:id => snipet.to_param, :snipet => invalid_attributes}
-        expect(response).to render_template("edit")
+    end
+    context "the snippet doesn't exist" do
+      before{expect(Snipet).to receive(:find).and_raise(ActiveRecord::RecordNotFound)}
+      it "redirect to all snippets" do
+        expect(delete :destroy, :id => generic_snippet.id).to redirect_to(:action => :index)
       end
     end
   end
 
-  describe "DELETE #destroy" do
-    it "destroys the requested snipet" do
-      sign_in build(:user)
-      snipet = Snipet.create! valid_attributes
-      expect {
-        delete :destroy, {:id => snipet.to_param}
-      }.to change(Snipet, :count).by(-1)
+  describe "PUT #give_snippet_a_star" do
+    let(:star){build_stubbed(:star)}
+    context "I didn't give the snippet a star yet" do
+      before do
+        expect(Star).to receive(:where).with(any_args).and_return(nil)
+        expect(put :give_snippet_a_star, :id => generic_snippet.id).to have_http_status(:ok)
+      end
+      it "create a new star" do
+        allow(Star).to receive(:create)
+      end
+      it "increment stars counter for this snippet" do
+      end
     end
-
-    it "redirects to the snipets list" do
-      sign_in build(:user)
-      snipet = Snipet.create! valid_attributes
-      delete :destroy, {:id => snipet.to_param}
-      expect(response).to redirect_to(snipets_path)
+    context "I already gave the snippet a star" do
+      before do
+        
+        expect(Star).to receive(:where).and_return(star)
+        expect(put :give_snippet_a_star, :id => generic_snippet.id).to have_http_status(:ok)
+        expect(User).to receive(:destroy).and_return(nil)
+      end
+      it "remove the star" do
+        expect(star).to receive(:destroy).and_return(nil)
```

Comment:
You could stub destroy method on "star" in some before filter, like:

```
expect(star).to receive(:destroy)
```
And here to set expectation, with that in place, destroy should not trigger actual action against db.